minigalaxy/installer.py

```diff
@@ -167,6 +170,35 @@ def copy_thumbnail(game):
     return error_message
 
 
+def create_applications_file(game):
+    error_message = ""
+    preferences_switch = Config.get("create_applications_file")
+    if game.platform == "linux" and preferences_switch:
+        path_to_shortcut = os.path.join(APPLICATIONS_DIR, game.name+".desktop")
+        # Create desktop file definition
+        desktop_context = {
+            "game_bin_path": os.path.join(game.install_dir, 'start.sh'),
+            "game_name": game.name,
+            "game_icon_path": os.path.join(game.install_dir, 'support/icon.png')
+            }
+        desktop_definition = """\
+            [Desktop Entry]
+            Type=Application
+            Terminal=false
+            StartupNotify=true
+            Exec="{game_bin_path}"
```

Comment:
Hmm, I think we will need to alter it a little bit for more complex commands. Maybe:
"
Exec=bash -c "{game_bin_path}"
"
It will require some testing...

Reply:
Not sure about this as bash might not be installed. But then again, the list of supported OSes should have bash by default. I'll test it the way you mentioned and commit it if there are no issues.

Reply:
Added /bin/bash -c and a loop to properly escape whitespaces in the path.